### PR TITLE
Todo list 

### PR DESCRIPTION
**Pull request by Bornittah**
**features**

- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Bornittah,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby Checks are failing. Kindly ensure that **all** checks are passing.
 

**Stylelint error:**
image


 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _Invalid Code Review Request **does not count** into the code reviews limit._

File: src/modules/storage.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    list.innerHTML = str;
+  }
+
+  static removeTodo = (index) => {
+    const todoList = Todo.getTodo(index);
+    todoList.splice(index, 1);
+    localStorage.setItem('todo', JSON.stringify(todoList));
+    Todo.display();
+    Todo.updateIndex();
+  }
```
Comment (codepantha) [inline src/modules/storage.js]: - [ ] Currently, there is no way to delete a to-do item from your app. I see that you have implemented the method, but there is no delete button where this functionality is hooked. Please fix this 🙏 
image

File: src/modules/storage.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    const todoList = Todo.getTodo();
+    const newtodoList = [];
+    todoList.forEach((item) => {
+      if (item.completed === false) {
+        newtodoList.push(item);
+      }
+      localStorage.setItem('todo', JSON.stringify(newtodoList));
+      Todo.display();
+    });
+  }
```
Comment (codepantha) [inline src/modules/storage.js]: - [ ] When you delete a to-do item or in this case, clear completed to-dos, it is expected that all remaining items' indices are updated so they represent the current list order and are unique. Currently, the indexes are not being updated as they should. Please fix this 🙏 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUESTED CHANGES ♻️ 

## Hi @Bornittah 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights ⭐ 
✔️ Descriptive PR
✔️ Linter checks
✔️ Meaningful commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] I know you're all about the functionality right now and you probably aren't too bothered about the UX of your app, and I get it. However, I would strongly suggest that you try to improve the UI of your app because as it stands, it leaves a lot to be desired 🙏 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
+
+Todo.display();
+Todo.checkedTask();
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const description = inputField.value;
+  const todo = new Todo(description);
+  todo.addTodo();
+});
```
Comment (chrystalme) [inline src/index.js]: Submit action is not adding new Todo to your list as you can see in the video below 

https://user-images.githubusercontent.com/12239821/173189744-189e7bcc-2040-4ac4-808c-9c1fb36495f7.mp4


Kindly check that you are getting the right form and passing the action to it.

**Review by chrystalme — CHANGES_REQUESTED** [review]
## Hi @Bornittah, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 💡 
- Linters passing all tests ✔️ 
- Following Gitflow guidelines ✔️ 


## Suggested changes ♻️ 
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Also, you can reach me on [Twitter](https://twitter.com/afam_ifeanyi) or [Slack](https://microverse-community.slack.com/team/U019ZC99KFA). Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 3 above)
Comment (Bornittah) [reply src/index.js]: Thanks, Trying using the enter key to submit the form data.  It would be better if you review my project at once

Comment (chrystalme) [reply src/index.js]: Hello @Bornittah. I just tried using `Enter` to submit the form and it did submit. Kindly implement the submit using the button and submit another pull-request.

**Comment by Oxford-G** [comment]
Hi Bornittah, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Oxford-G — COMMENTED** [review]
Hi Bornittah 

Your project is complete!  There is nothing else to say other than... it's time to merge it.
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**